packages/cli/src/verify.ts

```diff
@@ -0,0 +1,80 @@
+import { forge, getRpcUrl } from "@latticexyz/common/foundry";
+import { Hex, createWalletClient, getCreate2Address, http } from "viem";
+import { salt } from "./deploy/common";
+import { ensureDeployer } from "./deploy/ensureDeployer";
+import { privateKeyToAccount } from "viem/accounts";
+import { MUDError } from "@latticexyz/common/errors";
+import { getWorldFactoryContracts } from "./deploy/ensureWorldFactory";
+import { Contract } from "./deploy/ensureContract";
+
+type VerifyOptions = {
+  foundryProfile?: string;
+  systems: Contract[];
+  modules: Contract[];
+  worldAddress?: Hex;
+};
+
+async function verifyContract(
+  foundryProfile: string | undefined,
+  deployerAddress: Hex,
+  contract: Contract,
+  rpc: string,
+  cwd?: string,
+) {
+  if (!contract.label) {
+    throw new MUDError("Need a label");
+  }
+
+  const system = getCreate2Address({ from: deployerAddress, bytecode: contract.bytecode, salt });
+
+  await forge(["verify-contract", system, contract.label, "--verifier", "sourcify", "--rpc-url", rpc], {
+    profile: foundryProfile,
+    cwd,
+  });
+}
+
+export async function verify({
+  foundryProfile = process.env.FOUNDRY_PROFILE,
+  systems,
+  modules,
+  worldAddress,
+}: VerifyOptions): Promise<void> {
+  const privateKey = process.env.PRIVATE_KEY as Hex;
+  if (!privateKey) {
+    throw new MUDError(
+      `Missing PRIVATE_KEY environment variable.
+Run 'echo "PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80" > .env'
+in your contracts directory to use the default anvil private key.`,
+    );
+  }
+
+  const rpc = await getRpcUrl(foundryProfile);
+
+  const client = createWalletClient({
+    transport: http(rpc),
+    account: privateKeyToAccount(privateKey),
+  });
+
+  const deployerAddress = await ensureDeployer(client);
```

Comment:
why do we have to deploy the deployer when verifying?

Reply:
We should just be getting the address, oops

Reply:
we could just export `deployer` from `ensureDeployer` no? looks like it's constant

Reply:
Oh yes 🤔  Unless their chain does not allow for non EIP-155 signed transactions, which means the deployer is non-deterministic

Reply:
prob need to add `deployerAddress`